src/fios.cpp

```diff
@@ -745,3 +746,49 @@ void ScanScenarios()
 {
 	_scanner.Scan(true);
 }
+
+/**
+ * Constructs FiosNumberedSaveName. Initial number is the most recent save, or -1 if not found.
+ * @param prefix The prefix to use to generate a filename.
+*/
+FiosNumberedSaveName::FiosNumberedSaveName(const std::string &prefix) : prefix(prefix), number(-1)
+{
+	static std::string _prefix;
+	static std::optional<std::string> _autosave_path;
+	if (!_autosave_path) _autosave_path = FioFindDirectory(AUTOSAVE_DIR);
+
+	/* Callback for FiosFileScanner. */
+	static fios_getlist_callback_proc *proc = [](SaveLoadOperation fop, const std::string &file, const char *ext, char *title, const char *last) {
+		if (strcasecmp(ext, ".sav") == 0 && StrStartsWith(file, _prefix)) return FIOS_TYPE_FILE;
+		return FIOS_TYPE_INVALID;
+	};
+
+	/* Prefix to check in the callback. */
+	_prefix = *_autosave_path + this->prefix;
+
+	/* Get the save list. */
+	FileList list;
+	FiosFileScanner scanner(SLO_SAVE, proc, list);
+	scanner.Scan(".sav", _autosave_path->c_str(), false);
+
+	/* Find the number for the most recent save, if any. */
+	if (list.begin() != list.end()) {
+		SortingBits order = _savegame_sort_order;
+		_savegame_sort_order = SORT_BY_DATE | SORT_DESCENDING;
+		std::sort(list.begin(), list.end());
+		_savegame_sort_order = order;
+
+		std::string_view name = list.begin()->title;
+		std::from_chars(name.data() + this->prefix.size(), name.data() + name.size(), this->number);
+	}
```

Comment:
This is a lot of code for in a ctor. Somehow that triggers me. But I do not have any argument for it :P So it might be bullocks. Opinions? :D

Reply:
Well, the thing has to be initialized *somewhere* and where else should it be, if not in the constructor? And the code seems fairly specific to this one particular problem, so it's not like refactoring anything into separate functions would have much of an advantage.